zip-0208.rst

```diff
@@ -0,0 +1,224 @@
+::
+
+  ZIP: 208
+  Title: Shorter Block Target Spacing
+  Author: Simon Liu <simon@z.cash>
+          Daira Hopwood <daira@z.cash>
+  Category: Consensus
+  Created: 2018-01-10
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST", "SHOULD", "SHOULD NOT", and "MAY" in this document are to
+be interpreted as described in RFC 2119. [#RFC2119]_
+
+The terms "block chain", "consensus rule change", "branch", and "network
+upgrade" are to be interpreted as defined in [#zip-0200]_.
+
+The term "block target spacing" means the time interval between blocks targeted
+by the difficulty adjustment algorithm in a given branch. It is normally
+measured in seconds. (This is also sometimes called the "target block time",
+but "block target spacing" is the term used in [#latest-protocol]_.)
+
+
+Abstract
+========
+
+This proposal specifies a change in the block target spacing, to take effect in
+the Blossom network upgrade [#zip-0206]_.
+
+The emission schedule of mined ZEC will be approximately the same in terms of
+time, but this requires the emission per block to be adjusted to take account
+of the changed block target spacing.
+
+
+Motivation
+==========
+
+The motivations for decreasing the block target spacing are:
+
+- Reduced latency for considering transactions to be sufficiently confirmed;
+- Greater throughput of transactions in unit time.
+
+The latter goal could alternatively be achieved by increasing the block size
+limit, but that would not also achieve the former goal.
+
+Note that, for a given security requirement (in terms of the expected cost
+distribution of a rollback attack), the number of confirmations needed
+increases more slowly than the decrease in block time, and so, up to a point,
+decreasing the block target spacing can provide a better trade-off between
+latency and security. This argument assumes that the verification and
+propagation time for a block remain small compared to the block target spacing.
+
+
+Specification
+=============
+
+The following constants are defined:
+
+- PreBlossomPoWTargetSpacing := 150 seconds
+- PostBlossomPoWTargetSpacing := 75 seconds
+
+TODO: check that PostBlossomPoWTargetSpacing is reasonable.
```

Comment:
[Meta - these considerations are about issues that should be resolved before this or other TODO's can be resolved. Maybe it would be better to merge the PR with outstanding TODOs and the ZIP still in a draft state, and then use future PRs or tickets to track subsequent changes to the proposal. Thoughts?]

For `PostBlossomPoWTargetSpacing` there are two bounds:

- The lower (smaller interval) bound is defined by security / performance analysis.
- The upper (larger interval) bound is defined by "product considerations" as to whether or not the change is even worth it. Let's call this the "Is It Worth It" bound.

*As of this writing* the upper Is It Worth It bound is 75 seconds (half of the current target interval). In other words, if we determine for other reasons that an interval 75 seconds or lower is not feasible, then we should withdraw this proposal from NU2-Blossom upgrade.

We might also decide that this upper bound should change. For instance, we may later decide that if it's not at least 50 seconds or lower, then we should withdraw the proposal. (That number 50 is just a random example.)

**Potential Contention Alert:** What if we discover the lower bound is larger than the Is It Worth It upper bound *and* that is because the lower bound is predicated on security or performance parameters. We *MIGHT* decide to relax those security or performance parameters enough to meet the "Is It Worth It" bound. Doing so would be a *security degradation*!

So if we do that, let's be very explicit, clear, and loud about that fact. And remember, we may never hit that case.

However, a factor which would exacerbate this contention is the fact that currently the Is It Worth It bound is entirely arbitrary, and I just picked 75 on the notion that < 50% improvement isn't worth the new complexity. So if we enter that region of contention, we should put more effort into justifying the upper bound such as by basing it on some methodology or model of the benefit.